pkg/server/container_create_unix.go

```diff
@@ -182,11 +182,15 @@ func (c *criService) containerSpec(id string, sandboxID string, sandboxPid uint3
 	if !c.config.DisableProcMount {
 		// Apply masked paths if specified.
 		// If the container is privileged, this will be cleared later on.
-		specOpts = append(specOpts, oci.WithMaskedPaths(securityContext.GetMaskedPaths()))
+		if maskedPaths := securityContext.GetMaskedPaths(); maskedPaths != nil {
+			specOpts = append(specOpts, oci.WithMaskedPaths(maskedPaths))
```

Comment:
Wondering if it would make sense to adjust `obi.WithMaskedPaths()` to treat `nil` as `don't modify` https://github.com/containerd/containerd/blob/62083eeed275db8c154dd87a5f84444e97ddb507/oci/spec_opts.go#L931-L934

That would be a change in behaviour for that function of course (and would no longer allow resetting previously set values, although I'm not sure if that's a realistic use). Option was original added as part of https://github.com/containerd/containerd/pull/2269

So let me /cc @justincormack  as well

Reply:
`oci.WithMaskedPaths()`  is also used to reset a previously set value here.
https://github.com/containerd/containerd/blob/2991bc3de7fd5438577a14570f0115aff38065fd/oci/spec_opts.go#L1130-L1131

I found only these two occurrences of `oci.WithMaskedPaths()` in conteinerd, so changing the behavior may be acceptable, but I am not sure whether it is used outside of containerd.

Reply:
Thanks! I recall a conversation we had on https://github.com/moby/moby/pull/36644 (not sure if it's recorded on GitHub), where we were discussing that `nil` / `unset` meant "keep defaults", and passing an empty slice would mean `no masked paths`.

I agree that changing the `oci.WithMaskedPaths()` behavior is risky; current change looks good for now, but perhaps a follow-up discussion would be good 🤔